### PR TITLE
Remove most `waitForTimeout` usage from the freetext editor integration tests

### PR DESCRIPTION
**Pull request by timvandermeij**
The commit messages contain more details about the individual changes.

Fixes a part of #17656.

### CODE REVIEW

File: test/integration/freetext_editor_spec.mjs [hunk 1]
```diff
@@ -1364,9 +1361,6 @@ describe("FreeText Editor", () => {
           // Enter in editing mode.
           await switchToFreeText(page);
 
-          // eslint-disable-next-line no-restricted-syntax
-          await waitForTimeout(200);
```
Comment (timvandermeij) [inline test/integration/freetext_editor_spec.mjs:1368]: Note that `switchToFreeText` waits for the switching to be complete, see https://github.com/mozilla/pdf.js/blob/master/test/integration/freetext_editor_spec.mjs#L84, so this occurrence can therefore be removed without replacement.

File: test/integration/freetext_editor_spec.mjs [hunk 2]
```diff
@@ -56,9 +56,6 @@ const copyPaste = async page => {
   await kbCopy(page);
   await promise;
 
-  // eslint-disable-next-line no-restricted-syntax
-  await waitForTimeout(10);
```
Comment (timvandermeij) [inline test/integration/freetext_editor_spec.mjs:60]: Note that for both copy and paste we already wait for the corresponding event (above and below this block), so this occurrence can therefore be removed without replacement.

**Comment by timvandermeij** [comment]
The Windows failure above is already tracked in #17931 and therefore unrelated to this PR.

**Review by calixteman — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**